core/dbt/include/global_project/macros/adapters/common.sql

```diff
@@ -51,6 +51,29 @@
 
 {% endmacro %}
 
+{% macro get_create_index_sql(relation, index_dict) -%}
+  {{ return(adapter.dispatch('get_create_index_sql')(relation, index_dict)) }}
+{% endmacro %}
+
+{% macro default__get_create_index_sql(relation, index_dict) -%}
+  {{ exceptions.raise_not_implemented(
+    'get_create_index_sql macro not implemented for adapter '+adapter.type()) }}
```

Comment:
Let's just have this return `None`, rather than raise an exception. Then, `create_indexes` can run the query if and only if the SQL returned is not `None`. So by default, nothing happens; on Postgres, the appropriate SQL is returned and run.